File: src/main/java/cz/solutions/cockroach/RsuReport.java
package cz.solutions.cockroach;

import lombok.Value;

import java.util.List;
import java.util.Map;

@Value
public class RsuReport {
    List<PrintableRsu> printableRsuList;
    double rsuCroneValue;
    double rsuDollarValue;
    double taxableRsuCroneValue;
    int totalAmount;

    public Map<String, ?> asMap() {
        return Map.of(
                "rsuList", printableRsuList,
                "rsuCroneValue", FormatingHelper.formatDouble(rsuCroneValue),
                "rsuDolarValue", FormatingHelper.formatDouble(rsuDollarValue),
                "totalAmount", totalAmount,
                "taxableRsuCroneValue", FormatingHelper.formatDouble(taxableRsuCroneValue)
        );
    }
}

File: src/main/kotlin/cz/solutions/cockroach/RsuReport.kt
package cz.solutions.cockroach

data class RsuReport(
    val printableRsuList: List<PrintableRsu>,
    val rsuCroneValue: Double,
    val rsuDollarValue: Double,
    val taxableRsuCroneValue: Double,
    val totalAmount: Int
) {
    fun asMap(): Map<String, Any> {
        return mapOf(
            "rsuList" to printableRsuList,
            "rsuCroneValue" to FormatingHelper.formatDouble(rsuCroneValue),
            "rsuDolarValue" to FormatingHelper.formatDouble(rsuDollarValue),
            "totalAmount" to totalAmount,
            "taxableRsuCroneValue" to FormatingHelper.formatDouble(taxableRsuCroneValue)
        )
    }
}